cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/migration/ExampleMigrationVmWhenCpuMetricIsViolated.java

```diff
@@ -0,0 +1,340 @@
+        for (int i = 0; i < NUMBER_OF_HOSTS_TO_CREATE; i++) {
+            hostList.add(createHost(i, HOST_NUMBER_OF_PES, HOST_MIPS_BY_PE));
+            Log.printConcatLine("#Created host ", i, " with ", HOST_MIPS_BY_PE, " mips x ", HOST_NUMBER_OF_PES);
+        }
+        Log.printLine();
+
+        DatacenterCharacteristics characteristics
+                = new DatacenterCharacteristicsSimple(hostList)
+                .setCostPerSecond(DATACENTER_COST_PER_CPU)
+                .setCostPerMem(DATACENTER_COST_PER_RAM)
+                .setCostPerStorage(DATACENTER_COST_PER_STORAGE)
+                .setCostPerBw(DATACENTER_COST_PER_BW);
+
+        PowerVmAllocationPolicyMigrationWorstFitStaticThreshold allocationPolicy
+                = new PowerVmAllocationPolicyMigrationWorstFitStaticThreshold(
+                        new PowerVmSelectionPolicyMinimumUtilization(),
+                        overCpuUtilizationThreshold);
+        allocationPolicy.setUnderUtilizationThreshold(underCpuUtilizationThreshold);
+
+        PowerDatacenter dc = new PowerDatacenter(simulation, characteristics, allocationPolicy);
+        dc.setMigrationsEnabled(true).setSchedulingInterval(SCHEDULE_TIME_TO_PROCESS_DATACENTER_EVENTS);
+        return dc;
+    }
+
+    /**
+     *
+     * @param id
+     * @param numberOfPes
+     * @param mipsByPe
+     * @return
+     *
+     * @todo @author manoelcampos Using the {@link VmSchedulerSpaceShared} its
+     * getting NullPointerException, probably due to lack of CPU for all VMs. It
+     * has to be created an IT test to check this problem.
+     *
+     * @todo @author manoelcampos The method
+     * {@link DatacenterBroker#getCloudletsFinishedList()} returns an empty list
+     * when using null     {@link PowerDatacenter},
+     * {@link PowerHost} and {@link PowerVm}.
+     */
+    public static PowerHostUtilizationHistory createHost(int id, int numberOfPes, double mipsByPe) {
+        List<Pe> peList = createPeList(numberOfPes, mipsByPe);
+        PowerHostUtilizationHistory host = new PowerHostUtilizationHistory(id, HOST_STORAGE, peList);
+        host.setPowerModel(new PowerModelLinear(1000, 0.7))
+                .setRamProvisioner(new ResourceProvisionerSimple(new Ram(HOST_RAM)))
+                .setBwProvisioner(new ResourceProvisionerSimple(new Bandwidth(HOST_BW)))
+                .setVmScheduler(new VmSchedulerTimeShared());
+        return host;
+    }
+
+    public static List<Pe> createPeList(int numberOfPEs, double mips) {
+        List<Pe> list = new ArrayList<>(numberOfPEs);
+        for (int i = 0; i < numberOfPEs; i++) {
+            list.add(new PeSimple(i, new PeProvisionerSimple(mips)));
+        }
+        return list;
+    }
+
+    //We strongly encourage users to develop their own broker policies, to submit vms and cloudlets according
+    //to the specific rules of the simulated scenario
+    private static DatacenterBroker createBroker() {
```

Comment:
Since creating a broker is very simple with CloudSim Plus, remove this method createBroker and call the DatacenterBrokerSimple constructor directly where the createBroker method was being called.